src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents = JSON.parse(JSON.stringify(students));
+  const reducer = (acc: number, curt: number): number => acc + curt;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  sortedStudents.sort((a: Student, b: Student): number => {
+    let comparator: number = 0;
+
+    switch (sortBy) {
+      case ('name'):
+      case ('surname'):
+        comparator = a[sortBy].localeCompare(b[sortBy]);
+        break;
+      case ('age'):
+      case ('married'):
+        comparator = +a[sortBy] - +b[sortBy];
```

Comment:
can you do this without comparator?

```suggestion
        return +a[sortBy] - +b[sortBy]
```

Reply:
I can, but now I do this once `return (order === 'asc') ? comparator : comparator * -1;`
and without `comparator` I'll need to do this in each case
`return (order === 'asc') ? (+a[sortBy] - +b[sortBy]) : (+a[sortBy] - +b[sortBy]) * -1;`

So what is better?

Reply:
Let's think
Why should the code continue if we have already performed the action we need?
Why should we go through further checks?